h/static/scripts/annotator/plugin/pdf-metadata.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/**
+ * This PDFMetadata service provides implementations for extracting fields
+ * of a pdf document from the input PDFViewerApplication object.
+ *
+ * The fields extracted are:
+ * #1. urn - a string which uniquely identifies a pdf document.
+ * #2. title - a string representing the title of the pdf document.
+ * #3. link - an array of two objects, the urn and the url strings of the pdf.
+ * #4. documentFingerprint - the string representing the pdf fingerprint.
+ */
+function PDFMetadata(PDFViewerApplication) {
+  this._PDFViewerApplication = PDFViewerApplication;
+  this._documentLoaded = new Promise(function (resolve) {
+    var finish;
+    finish = function (evt) {
+      window.removeEventListener('documentload', finish);
+      return resolve();
+    };
+    return window.addEventListener('documentload', finish);
+  });
+};
+
+/**
+ * Returns a promise of a PDF URN string after the pdf loads
+ * succesfully and the 'documentload' event fires.
+ */
+PDFMetadata.prototype.getUrn = function () {
+  return this._documentLoaded.then(function () {
+    return 'urn:x-pdf:' + this._PDFViewerApplication.documentFingerprint;
+  }.bind(this));
+};
+
+/**
+ * Returns a promise of an object with the fields
+ * title(string), link(array) and documentFingerprint(string)
+ * after the pdf loads succesfully and the 'documentload' event fires.
+ */
+PDFMetadata.prototype.getMetadata = function () {
+  return this._documentLoaded.then(function () {
+    var info;
+    var metadata;
+    var urn;
+    var title;
+    var documentFingerprint;
+    var link = [];
+
+    info = this._PDFViewerApplication.documentInfo;
+    metadata = this._PDFViewerApplication.metadata;
+
+    if ((typeof metadata !== 'undefined' && metadata !== null ? metadata.has('dc:title') : void 0) &&
+      metadata.get('dc:title') !== 'Untitled') {
+      title = metadata.get('dc:title');
+    } else if (typeof info !== 'undefined' && info !== null ? info.Title : void 0) {
+      title = info.Title;
+    } else {
+      title = document.title;
+    }
+
+    urn = 'urn:x-pdf:' + this._PDFViewerApplication.documentFingerprint;
+    link = [{href: urn}, {href: this._PDFViewerApplication.url}];
+    documentFingerprint = this._PDFViewerApplication.documentFingerprint;
+
+    return {
+      'title': title,
```

Comment:
Nit: Keys in object literals should only be quoted if they contain special characters that are not legal in field names.